keras/utils/layer_utils.py

```diff
@@ -24,12 +24,29 @@ def print_summary(model, line_length=None, positions=None, print_fn=print):
         sequential_like = True
     else:
         sequential_like = True
-        for v in model.nodes_by_depth.values():
+        nodes_by_depth = model.nodes_by_depth.values()
+        nodes = []
+        for v in nodes_by_depth:
             if (len(v) > 1) or (len(v) == 1 and len(v[0].inbound_layers) > 1):
                 # if the model has multiple nodes or if the nodes have multiple inbound_layers
                 # the model is no longer sequential
                 sequential_like = False
                 break
+            nodes += v
+        if sequential_like:
+            # search for shared layers
+            for layer in model.layers:
+                flag = False
+                for node in layer.inbound_nodes:
+                    if node in nodes:
```

Comment:
Wouldn't this always be True?

Reply:
Not necessarily. A layer might be shared between 2 models and hence have more than one inbound node. But we are looking for layers which are being shared within the same model.

Reply:
```python
dense = Dense(10)
a = Input((10,))
b = dense(a)
model1 = Model(a, b)

a2 = Input((10,))
b2 = dense(a2)
model2 = Model(a2, b2)

model1.summary()  
# Here dense layer has 2 inbound nodes, 
#but the second node is not part of model1.
# So should be considered sequential like